src/services/identity/__tests__/UsersService.spec.ts

```diff
@@ -120,71 +121,31 @@ describe("User Service", () => {
     expect(actual.githubId).toBe(mockGithubId)
   })
 
-  it("should allow whitelisted emails", async () => {
-    // Arrange
-    const expected = true
-    const mockWhitelistEntry = {
-      email: ".gov.sg",
-    }
-    MockWhitelist.findAll.mockResolvedValueOnce([mockWhitelistEntry])
-
-    // Act
-    const actual = await UsersService.canSendEmailOtp(mockEmail)
-
-    // Assert
-    expect(actual).toBe(expected)
-  })
-  it("should not allow partial match of whitelisted emails", async () => {
-    // Arrange
-    const expected = false
-    // NOTE: This ends with gov.sg not .gov.sg (lacks a dot after the @)
-    const emailWithoutDot = "user@gov.sg"
-    const mockWhitelistEntry = {
-      email: ".gov.sg",
-    }
-    MockWhitelist.findAll.mockResolvedValueOnce([mockWhitelistEntry])
-
-    // Act
-    const actual = await UsersService.canSendEmailOtp(emailWithoutDot)
-
-    // Assert
-    expect(actual).toBe(expected)
-  })
-
-  it("should allow not .gov.sg emails when whitelist does not contain .gov.sg", async () => {
-    // Arrange
-    const expected = false
-    const mockWhitelistEntries = [
-      {
-        email: "@accenture.com",
-      },
-      {
-        email: ".edu.sg",
-      },
-    ]
-    MockWhitelist.findAll.mockResolvedValueOnce(mockWhitelistEntries)
+  describe("canSendEmailOtp", () => {
+    it("should return true when the db query returns a record", async () => {
+      // Arrange
+      const expected = true
+      MockSequelize.query.mockResolvedValueOnce([{ email: ".gov.sg" }])
 
-    // Act
-    const actual = await UsersService.canSendEmailOtp(mockEmail)
+      // Act
+      const actual = await UsersService.canSendEmailOtp(mockEmail)
 
-    // Assert
-    expect(actual).toBe(expected)
-  })
+      // Assert
+      expect(actual).toBe(expected)
+    })
 
-  it("should not allow suffix match if the whitelist entry is a full email", async () => {
-    // Arrange
-    const expected = false
-    const mockWhitelistEntries = [
-      {
-        email: "foo@accenture.com",
-      },
-    ]
-    MockWhitelist.findAll.mockResolvedValueOnce(mockWhitelistEntries)
+    it("should return false when the db query returns no record", async () => {
+      // Arrange
+      const expected = false
+      // NOTE: This ends with gov.sg not .gov.sg (lacks a dot after the @)
+      const emailWithoutDot = "user@gov.sg"
+      MockSequelize.query.mockResolvedValueOnce([])
```

Comment:
I think we should still be adding the mock for `[{ email: ".gov.sg" }]` - our test suite clears the db state between tests, so this test is checking for `user@gov.sg` against an empty whitelist

Reply:
Oh wait what? the tests can populate the DB? I would love indeed  to run the query for real rather than arbitrarily mocking a db response!

Reply:
Whoops sorry i meant the jest mocks are reset! So we should be mocking the return of query here as `[{ email: ".gov.sg" }] `

That said we do have population of DB for some of our tests in `integration` - we don't do this for the tests in each service which we consider unit tests

The `/integration/Users.spec.ts` test suite does cover some whitelist tests though (under the `/email/otp` suite), so if we want to test this against the db we could add this test there as well!

Reply:
Ah I see, I'll take a look at the integration tests (I could probably use some help for that 😅)

But so specifically in this case you are commenting on here, the mock does what we expect. For reference, the new query no longer return the full content of the table, it returns either:
- no match if the input matches none of the whitelist entries
OR
- the first whitelist entry the input matches

These 2 distinct response types represent a boolean-ish yes/no state of whether the input is whitelisted.

And in this case, the input `user@gov.sg` is not expecting to match `.gov.sg` as a whitelist entry, so that cannot be in the mocked response.

Reply:
Ah you're right, I got the intent of the test wrong sorry! But yeah i think we can then introduce an integration test with this scenario (db contains `.gov.sg`, user tried to login via `blah@gov.sg` which should not match)

Reply:
I've updated the comments to make it clearer why a raw query is used: https://github.com/isomerpages/isomercms-backend/pull/1301/commits/db15356463310f164287a44132d39c9eca2bd888